scripts/setup-updates

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python3
+
+import sys
+import os
+import pydbus
+
+def symlink_updates(dest_dir, update_dir):
+    """Setup symlinks for the updates from the updates image.
+
+    :param str dest_dir: symlink target
+    :param str update_dir: symlink source (updates image content)
+    """
+    contents = os.listdir(update_dir)
+
+    for f in os.listdir(dest_dir):
+        dest_path = os.path.join(dest_dir, f)
+        update_path = os.path.join(update_dir, f)
+        if f in contents:
+            # recurse into directories, there might be files missing in updates
+            if os.path.isdir(dest_path) and os.path.isdir(update_path):
+                symlink_updates(dest_path, update_path)
+        else:
+            if f.endswith(".pyc") or f.endswith(".pyo"):
+                continue
+            os.symlink(dest_path, update_path)
+
+def setup_python_updates():
+    """Setup updates to Anaconda Python files."""
+    from distutils.sysconfig import get_python_lib
+    import gi.overrides
+
+    if "ANACONDA_WIDGETS_OVERRIDES" in os.environ:
+        for p in os.environ["ANACONDA_WIDGETS_OVERRIDES"].split(":"):
+            gi.overrides.__path__.insert(0, os.path.abspath(p))
+
+    # Temporary hack for F18 alpha to symlink updates and product directories
+    # into tmpfs.  To be removed after beta in order to directly use content
+    # from /run/install/ -- JLK
+    for dirname in ("updates", "product"):
+        if os.path.exists("/run/install/%s" % dirname):
+            if os.path.islink("/tmp/%s" % dirname):
+                # Assume updates have already been setup
+                return
+            os.symlink("/run/install/%s" % dirname,
+                       "/tmp/%s" % dirname)
```

Comment:
Can't we remove it now, when it was just a temporal hack which should be removed after beta?

Reply:
Unfortunately it doesn't seem to work without the temporary hack - I assume something is using the the old & something the new location. But definitely something for a followup investigation & PR.